### ISSUE TITLE
self, crate, super, Self are incorrectly escaped.

### ISSUE THREAD

**Issue by noclue** [JoelCourtney/check_keyword#4]
```
error: `self` cannot be a raw identifier
     --> src/output.rs:37296:9
      |
37296 |     pub r#self: Option<Box<Reference>>,
      |         ^^^^^^
```

Nice library! I love the usability.

However it does not escape correctly all identifiers. Note that self, crate, super, Self cannot be treated as raw identifiers i.e. escaped with r#.

See https://doc.rust-lang.org/reference/identifiers.html

I suggest you suffix these with `_`. A suffix would not have side effect of suppressing unused warnings in generated code.

**Comment by JoelCourtney** [comment]
Thanks! Good suggestion

**Comment by JoelCourtney** [comment]
I've published `0.3.0` on crates.io. It includes proper escapes for identifiers that can't be raw, and also adds support for weak keywords. Thanks!
